checks/smtp/smtp.go

```diff
@@ -0,0 +1,18 @@
+package smtp
+
+import (
+	"context"
+	"fmt"
+	"net/smtp"
+)
+
+// New creates a new SMTP health check function.
+func TestConnection(client *smtp.Client) func(ctx context.Context) error {
```

Comment:
We should probably change this to `New` as well to stick to the norm.
 ```suggestion
func New(client *smtp.Client) func(ctx context.Context) error {
```